plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqCodeTest/CodeWhispererUTGChatManager.kt

```diff
@@ -493,14 +528,16 @@ class CodeWhispererUTGChatManager(val project: Project, private val cs: Coroutin
                 launchTestGenFlow(prompt, codeTestChatHelper, previousIterationContext, selectionRange)
             } catch (e: Exception) {
                 // Add an answer for displaying error message
-                var errorMessage = e.message
-                if (e is JsonParseException) {
-                    errorMessage = message("testgen.error.generic_error_message")
+                val errorMessage = when {
+                    e is CodeTestException && e.statusCode == "400" &&
+                        e.message?.startsWith("CreateTestJobError: Maximum") == true ->
```

Comment:
is there a better way of doing this

Reply:
Here Quota exceeded is subset of Throttling exception so opted for checking the error message.